### PR TITLE
LPS-166461 - Automation guest view permission

### PR DESCRIPTION
**Pull request by marcosvictor22**
@david-gutierrez-mesa

### CODE REVIEW

**Comment by marcosvictor22** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase [hunk 1]
```diff
@@ -1345,4 +1345,56 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-166461. There is view permission for the guest"
+	@priority = 4
+	test ViewForTheUserToFindThePage {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase:1350]: Rename to ViewContentPageShownInSearchResultsAfterRemoveViewPermissionForGuest
Move the case to ContentPagesWithSearch.testcase

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase [hunk 2]
```diff
@@ -1345,4 +1345,56 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-166461. There is view permission for the guest"
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase:1348]: Use bug ticket number LPS-163785

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase [hunk 3]
```diff
@@ -1345,4 +1345,56 @@ definition {
...
+				fragmentName = "Paragraph");
+
+			PageEditor.editFragmentText(
+				fragmentName = "Paragraph",
+				id = "element-text",
+				text = "lorem ipsum");
+
+			JSONLayout.publishLayout(
+				groupName = ${randomSiteName},
+				layoutName = "Test Page Name");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase:1374]: Use PageEditor.publish();

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase [hunk 4]
```diff
@@ -1345,4 +1345,56 @@ definition {
...
+		}
+
+		task ("When user shows page by text search") {
+			PagesAdmin.openPagesAdmin(siteURLKey = ${siteURLKey});
+
+			Search.searchCP(searchTerm = "lorem");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase:1381]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase [hunk 5]
```diff
@@ -1345,4 +1345,56 @@ definition {
...
+				pageName = "Test Page Name",
+				site = ${randomSiteName});
+
+			Role.removePermissionPG(
+				permissionDefinitionKey = "CONTENT_PERMISSIONS_VIEW_CHECKBOX",
+				roleTitle = "Guest");
+		}
+
+		task ("Then the user can still view the page by text search") {
+			PagesAdmin.openPagesAdmin(siteURLKey = ${siteURLKey});
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagestree/PagesTree.testcase:1394]: Navigate to the Search page

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @marcosvictor22, could you refactor according to comments? Thanks

**Comment by marcosvictor22** [comment]
@Tim-Cao echo repository needs to be updated

**Comment by Tim-Cao** [comment]
Hi @marcosvictor22, could you double check if the changes you made are pushed?

**Comment by marcosvictor22** [comment]
I made the changes, but it didn't send, how strange... @Tim-Cao I'll do it again

**Comment by Tim-Cao** [comment]
See changes @marcosvictor22 https://github.com/liferay-echo/liferay-portal/pull/12981

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**